st2common/st2common/config.py

```diff
@@ -82,6 +82,14 @@ def register_opts(ignore_errors=False):
     ]
     do_register_opts(content_opts, 'content', ignore_errors)
 
+    webui_opts = [
+        cfg.StrOpt('webui_base_url', default='https://localhost',
```

Comment:
`localhost` is never a good option: nobody ever installs St2 on a local machine, and even if they would, it would only be valid for one user in a channel.

What about using the output of `/bin/hostname` as a default? Yes, it will still be wrong in a lot of cases, but still better than `localhost` and seems to be the least bad option overall.